src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,25 +550,63 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             Dim Here = 1
-            While CanGet(Here)
-                ch = Peek(Here)
-                If IsWhitespace(ch) Then
-                    Here += 1
+            Dim atNewLine As Boolean
+            ' Code above is identical to V15.5
+            Dim FoundCommentAfterUnderscore As Boolean = False
+            ' Line continuation is valid at the end of the line, or at the end of the file, or followed by a trailing comment.
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then
+                While CanGet(Here)
+                    ch = Peek(Here)
+                    If IsWhitespace(ch) Then
+                        Here += 1
+                    Else
+                        Exit While
+                    End If
+                End While
+                If CanGet(Here) AndAlso IsSingleQuote(Peek(Here)) Then
+                    ' You only get here if running V16 and have a comment
+                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                    If Here > 1 Then
+                        tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                    End If
+                    Dim comment As SyntaxTrivia = ScanComment()
+                    If Not CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                        If comment IsNot Nothing Then
+                            ' PROTOTYPE Need correct error here
+                            comment = DirectCast(comment.SetDiagnostics({ErrorFactory.ErrorInfo(ERRID.ERR_LineContWithCommentPreV161,
+                                                                                                New VisualBasicRequiredLanguageVersion(Feature.CommentsAfterLineContinuation.GetLanguageVersion()))}), SyntaxTrivia)
+                        End If
+                    End If
+                    tList.Add(comment)
+                    ch = Peek()
+                    atNewLine = IsNewLine(ch)
+                    FoundCommentAfterUnderscore = True
                 Else
-                    Exit While
+                    ' If you get her you have a Line Continuation without comment, so process as V15.5 below
                 End If
-            End While
-
-            ' Line continuation is valid at the end of the
-            ' line or at the end of file only.
-            Dim atNewLine = IsNewLine(ch)
-            If Not atNewLine AndAlso CanGet(Here) Then
-                Return False
             End If
+            If Not FoundCommentAfterUnderscore Then
```

Comment:
the logic here seems very contorted and difficult to follow.  i don't really understand why the flow is like it is.

Reply:
I feel like it would be much easier to just make the flow look like this:

1. Scan for the accepted patterns.  i.e.:

```
<underscore><newline>
<underscore><endoffile>
<underscore><optional_spaces><comment>
```

If it is none of those, return.

Otherwise:

1. consume the underscope.
2. check if you have spaces, if so consume spaces and comment.
3. if at newline, consume newline, then do that little special bit of logic at the end 

--

This would be very clear logic without the need to set all these intermediary variables and do conditional control flow.

Reply:
It was to leave the old code unchanged when the _ is not followed by a comment and a higher level expects adds the error ERR_LineContWithCommentOrNoPrecSpace. I looked at merging the two cases into the new code buy it seemed riskier.

Reply:
>  looked at merging the two cases into the new code but it seemed riskier.

we can address risk by adding enough tests.  we shoudl make this code clear and easy to understand.  otherwise the chance of breaking things in the future (or getting something wrong now), is much higher.

Reply:
I have restructured, it should be easier to understand. ScannerTest.VB already tests all the various options around comments, spaces and end of file. I might just need to add test for V15.5 vs new behavior. I will look at that soon.